include/dwfl_module.hpp

```diff
@@ -10,10 +10,21 @@ extern "C" {
 namespace ddprof {
 
 struct DDProfMod {
-  DDProfMod() : _mod(nullptr), _low_addr(0), _high_addr(0) {}
+  enum Status {
```

Comment:
Haha, I really like this.  Usually a status would have a "Good" type, but since we _cannot_ make such an evaluation (if we could, then because it's all happening in a separate process, our good view of the world can immediately become bad), then the best we can say is we haven't discovered an inconsistency _yet_.

Am I reading the semantics correctly?

Reply:
Exactly my thoughts :-)